hr_employee_firstname/models/hr_employee.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2010-2014 Savoir-faire Linux (<http://www.savoirfairelinux.com>)
+# Copyright 2016-2019 Onestein (<https://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+UPDATE_PARTNER_FIELDS = ['firstname', 'lastname', 'user_id', 'address_home_id']
+
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    @api.model
+    def _get_name(self, lastname, firstname):
+        return self.env['res.partner']._get_computed_name(lastname, firstname)
+
+    @api.onchange('firstname', 'lastname')
+    def _onchange_firstname_lastname(self):
+        if self.firstname and self.lastname:
+            self.name = self._get_name(self.lastname, self.firstname)
+
+    def _firstname_default(self):
+        return ' ' if self.env.context.get('module') else False
```

Comment:
Better extract default firstname to a variable and reuse it (at least 2 places of magic constant use)